src/purifier/purifier.js

```diff
@@ -17,11 +22,66 @@ module.exports = purgeContentObjects = async (apiKey, internal = 0) => {
                 i++;
             }
         } else {
-            i--;
+            i = 0;
+            if (contentTypeDefinitions.length !== ctdArrFormerLength) {
+                ctdArrFormerLength = contentTypeDefinitions.length
+            } else {
+                if (!force) {
+                    console.log("Purge command stumbled upon relation loop in CTDs:");
+                    while (i < contentTypeDefinitions.length) {
+                        console.log(contentTypeDefinitions[i].name)
+                        i++;
+                    }
+                    console.log("Use `flotiq purge [apiKey] --force` or remove conflicting relations manually");
+                    return;
+                } else {
+                    spinner = ora(`Cleaning data of relation loops, please do not stop the command or close the terminal... Content Types cleared of looped relations: ${ctdsClearedOfRelations}\n`).start();
+                    await dropRelations(contentTypeDefinitions.slice(ctdsClearedOfRelations), apiKey);
+                    ctdsClearedOfRelations++;
+                    spinner.stop();
+                }
+            }
+        }
+    }
+}
+
+const dropRelations = (contentTypeDefinitions, apiKey) => {
+    const removeProperty = (ctd, property) => {
+        delete ctd.metaDefinition.propertiesConfig[property];
+        delete ctd.schemaDefinition.allOf[1].properties[property];
+        ctd.metaDefinition.order.splice(ctd.metaDefinition.order.indexOf(property), 1);
+        if (ctd.schemaDefinition.required.includes(property)) {
+            ctd.schemaDefinition.required.splice(ctd.schemaDefinition.required.indexOf(property), 1);
+        }
+        return ctd;
+    }
+
+    const cloneObject = (obj) => {
+        return JSON.parse(JSON.stringify(obj));
+    }
+
+    const clearContentType = async (ctd) => {
+        let ctdWithDroppedRelations = cloneObject(ctd);
+        for (let property in ctd.metaDefinition.propertiesConfig) {
+            if (ctd.metaDefinition.propertiesConfig[property]?.validation?.hasOwnProperty("relationContenttype")) {
```

Comment:
In complex if statements I suggest create const variable before if, and name this variable
```
const isRelationField = ctd.metaDefinition.propertiesConfig[property]?.validation?.hasOwnProperty("relationContenttype");

if(isRelationField) {...
```